sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ServiceBusSessionsEndToEndTests.cs

```diff
@@ -264,31 +275,40 @@ public async Task ServiceBusSessionSub_SessionLocks()
                 Assert.True(_waitHandle1.WaitOne(SBTimeoutMills));
                 Assert.True(_waitHandle2.WaitOne(SBTimeoutMills));
 
-                IEnumerable<LogMessage> logMessages1 = host.GetTestLoggerProvider().GetAllLogMessages();
-                Assert.False(logMessages1.Where(p => p.Level == LogLevel.Error).Any());
+                var logMessages = GetLogMessages(host)
+                    .Where(m => m.Category == "Function.SBSub1Trigger.User").ToList();;
 
-                // filter out anything from the custom processor for easier validation.
-                List<LogMessage> consoleOutput1 = logMessages1.Where(m => m.Category == "Function.SBSub1Trigger.User").ToList();
-                Assert.True(consoleOutput1.Count() == 10, ServiceBusSessionsTestHelper.GetLogsAsString(consoleOutput1));
-                double seconds = (consoleOutput1[5].Timestamp - consoleOutput1[4].Timestamp).TotalSeconds;
+                Assert.True(logMessages.Count() == 10, ServiceBusSessionsTestHelper.GetLogsAsString(logMessages));
+                double seconds = (logMessages[5].Timestamp - logMessages[4].Timestamp).TotalSeconds;
                 Assert.True(seconds > 90 && seconds < 110, seconds.ToString());
-                for (int i = 0; i < consoleOutput1.Count(); i++)
+                for (int i = 0; i < logMessages.Count(); i++)
                 {
                     if (i < 5)
                     {
-                        Assert.AreEqual(consoleOutput1[i].FormattedMessage[consoleOutput1[0].FormattedMessage.Length - 1],
-                            consoleOutput1[0].FormattedMessage[consoleOutput1[0].FormattedMessage.Length - 1]);
+                        Assert.AreEqual(logMessages[i].FormattedMessage[logMessages[0].FormattedMessage.Length - 1],
+                            logMessages[0].FormattedMessage[logMessages[0].FormattedMessage.Length - 1]);
                     }
                     else
                     {
-                        Assert.AreEqual(consoleOutput1[i].FormattedMessage[consoleOutput1[0].FormattedMessage.Length - 1],
-                            consoleOutput1[5].FormattedMessage[consoleOutput1[0].FormattedMessage.Length - 1]);
+                        Assert.AreEqual(logMessages[i].FormattedMessage[logMessages[0].FormattedMessage.Length - 1],
+                            logMessages[5].FormattedMessage[logMessages[0].FormattedMessage.Length - 1]);
                     }
                 }
+
                 await jobHost.StopAsync();
             }
         }
 
+        private static List<LogMessage> GetLogMessages(IHost host)
+        {
+            List<LogMessage> logMessages = host.GetTestLoggerProvider().GetAllLogMessages()
+                // Filter out Azure SDK and custom processor logs for easier validation. Intentionally do the error check after as the
+                // Service Bus SDK currently logs errors when stopping the processor.
```

Comment:
Do we have a work item to fix this?

Reply:
Yes - https://github.com/Azure/azure-sdk-for-net/issues/19098#issuecomment-813126915

Reply:
Link it from the comment please.